src/monitor/v2_queue_monitor.rs

```
@@ -0,0 +1,409 @@
+                .limit(limit)
+        }
+        QueueType::OutputStateV2 => {
+            accounts::Entity::find()
+                .filter(accounts::Column::Tree.eq(tree_bytes))
+                .filter(accounts::Column::LeafIndex.gte(start_offset as i64))
+                .filter(accounts::Column::LeafIndex.lt((start_offset + limit) as i64))
+                .order_by_asc(accounts::Column::LeafIndex)
+                .limit(limit)
+        }
+        QueueType::AddressV2 => {
+            accounts::Entity::find()
+                .filter(accounts::Column::Tree.eq(tree_bytes))
+                .filter(accounts::Column::LeafIndex.gte(start_offset as i64))
+                .filter(accounts::Column::LeafIndex.lt((start_offset + limit) as i64))
+                .order_by_asc(accounts::Column::LeafIndex)
+                .limit(limit)
+        }
+        _ => {
+            return Err(PhotonApiError::ValidationError(format!(
+                "Unsupported queue type: {:?}",
+                queue_type
+            )))
+        }
+    };
+
+    let elements = query
+        .all(db)
+        .await
+        .map_err(|e| PhotonApiError::UnexpectedError(format!("DB error: {}", e)))?;
+
+    // Convert to hash values based on queue type
+    let mut result = Vec::with_capacity(elements.len());
+    for element in elements.iter() {
+        let mut value = [0u8; 32];
+        
+        match queue_type {
+            QueueType::InputStateV2 => {
+                // Input queues use nullifiers for hash chains
+                if let Some(ref nullifier) = element.nullifier {
+                    if nullifier.len() >= 32 {
+                        value.copy_from_slice(&nullifier[..32]);
+                    }
+                }
+            }
+            _ => {
+                // Output and Address queues use account hashes
+                if element.hash.len() >= 32 {
+                    value.copy_from_slice(&element.hash[..32]);
+                }
+            }
```

these should throw an error on else. Else should be unreachable but better be sure.